include/drivers/rtc.h

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file
+ * @brief RTC public API header file.
+ */
+/*
+ * Copyright (c) 2021 LACROIX Group
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <time.h>
+#include <device.h>
+// #include <dt-bindings/rtc/rtc.h>
+
+/**
+ * @brief RTC driver APIs
+ * @defgroup rtc_interface RTC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief Wake up timers enum. */
+enum rtc_wakeup_id {
+	RTC_WUT0,       /**< Wake Up timer 0. */
+	/* Add new item before RTC_WUT_NUM. */
+	RTC_WUT_NUM,    /**< Number of wut for the driver. */
+};
+
+/** @brief Alarms enum. */
+enum rtc_alarm_id {
+	RTC_ALARMA,     /**< Alarm A. */
+	RTC_ALARMB,     /**< Alarm B. */
+	/* Add new item before RTC_ALARM_NUM. */
+	RTC_ALARM_NUM,  /**< Number of alarms for the driver. */
+};
+
+/** @brief Specifies if the RTC Alarm is on date or Week day. */
+enum rtc_alarm_date_weekday {
+	RTC_ALARM_DATE_SEL,     /**< Alarm Date is selected, set parameter to a value in the 1-31 range. */
+	RTC_ALARM_WEEKDAY_SEL,  /**< Alarm Weekday is selected, set parameter to a value in the 0-6 range (days since Sunday). */
+};
+
+/** @brief Specifies the RTC Alarm Masks. */
+enum rtc_alarm_mask {
+	RTC_ALARM_MASK_NONE,            /**< No mask. */
+	RTC_ALARM_MASK_DATE_WEEKDAY,    /**< Mask on date or weekday. */
+	RTC_ALARM_MASK_HOURS,           /**< Mask on hours. */
+	RTC_ALARM_MASK_MIN,             /**< Mask on minutes. */
+	RTC_ALARM_MASK_SEC,             /**< Mask on seconds. */
+	RTC_ALARM_MASK_ALL,             /**< Mask on all. */
+};
+
+/** @brief Wake up timer structure. */
+struct rtc_wakeup {
+	uint32_t period;                                /**< Wake up timer period in milliseconds. */
+	void (*callback)(const struct device *dev);     /**< Callback function. Can be NULL. */
+};
+
+/** @brief Alarm structure.
+ * @param alarm_time Alarm time members
+ * @param alarm_date_weekday_sel Alarm selection: on Date or on WeekDay.
+ * 	If Alarm date is selected, use tm_mday [1 to 31] to set the alarm
+ *  If WeekDay date is selected, use tm_wday [0 to 6] to set the alarm
+ * @param callback	Callback function. Can be NULL.
+ */
+struct rtc_alarm {
+	struct tm alarm_time;           /**< Alarm time members. */
+	enum rtc_alarm_mask alarm_mask; /**< Alarm mask. */
+	/** Alarm selection: on Date or on WeekDay.
+	 * If Alarm date is selected, use tm_mday [1 to 31] to set the alarm
+	 * If WeekDay date is selected, use tm_wday [0 to 6] to set the alarm
+	 */
+	enum rtc_alarm_date_weekday alarm_date_weekday_sel;
+	void (*callback)(const struct device *dev);     /**< Callback function. Can be NULL. */
```

Comment:
See this comment https://github.com/zephyrproject-rtos/zephyr/pull/41287/files#r902898525